react/components/QuantitySelector.tsx

```diff
@@ -19,51 +23,88 @@ const messages = defineMessages({
 const MAX_DROPDOWN_VALUE = 10
 const MAX_INPUT_LENGTH = 5
 
-enum SelectorType {
-  Dropdown,
-  Input,
+/* eslint-disable no-shadow */
+const enum SelectorType {
+  Dropdown = 'Dropdown',
+  Input = 'Input',
 }
+/* eslint-enable no-shadow */
 
 interface Props {
-  id: string
+  id?: string
   value: number
   maxValue: number
   onChange: (value: number) => void
-  disabled: boolean
+  disabled?: boolean
+  unitMultiplier?: number
+  measurementUnit?: string
 }
 
-const normalizeValue = (value: number, maxValue: number) =>
-  value > maxValue ? maxValue : value
+const normalizeValue = (value: number, maxValue: number) => {
+  const normalizedValue = value > maxValue ? maxValue : value
+
+  return normalizedValue
+}
 
-const validateValue = (value: string, maxValue: number) => {
-  const parsedValue = parseInt(value, 10)
+const validateValue = (
+  value: string,
+  maxValue: number,
+  unitMultiplier: number
+) => {
+  const parsedValue = parseFloat(value.replace(',', '.'))
 
-  if (isNaN(parsedValue)) {
+  if (Number.isNaN(parsedValue)) {
     return 1
   }
 
-  return normalizeValue(parseInt(value, 10), maxValue)
+  return normalizeValue(Math.round(parsedValue / unitMultiplier), maxValue)
 }
 
-const validateDisplayValue = (value: string, maxValue: number) => {
-  const parsedValue = parseInt(value, 10)
+const validateDisplayValue = (
+  value: string,
+  maxValue: number,
+  unitMultiplier: number
+) => {
+  const parsedValue = parseFloat(value.replace(',', '.'))
 
-  if (isNaN(parsedValue) || parsedValue < 0) {
-    return ''
+  if (Number.isNaN(parsedValue) || parsedValue < 0) {
+    return 1
   }
 
-  return `${normalizeValue(parsedValue, maxValue)}`
+  const normalizedValue = normalizeValue(parsedValue, maxValue) * unitMultiplier
+
+  return normalizedValue
 }
 
-const getDropdownOptions = (maxValue: number, intl: InjectedIntl) => {
+const getDropdownOptions = ({
+  maxValue,
+  intl,
+  unitMultiplier,
+  measurementUnit,
+}: {
+  maxValue: number
+  intl: IntlShape
+  unitMultiplier: number
+  measurementUnit?: string
+}) => {
   const limit = Math.min(9, maxValue)
   const options = [
     { value: 0, label: `0 - ${intl.formatMessage(messages.remove)}` },
-    ...range(1, limit + 1).map(idx => ({ value: idx, label: `${idx}` })),
+    ...range(1, limit + 1).map(idx => ({
+      value: idx,
+      label: `${intl.formatNumber(idx * unitMultiplier)}${
+        measurementUnit ? ` ${measurementUnit}` : ''
+      }`,
+    })),
   ]
 
   if (maxValue >= MAX_DROPDOWN_VALUE) {
-    options.push({ value: MAX_DROPDOWN_VALUE, label: `${MAX_DROPDOWN_VALUE}+` })
+    options.push({
+      value: MAX_DROPDOWN_VALUE,
+      label: `${intl.formatNumber(MAX_DROPDOWN_VALUE * unitMultiplier)}${
+        measurementUnit ? ` ${measurementUnit}` : ''
```

Comment:
```suggestion
        measurementUnit || ''
```

Reply:
I actually need a _little_ space before the measurement unit :-/

I'll this to a message anyway, so this should be fine